ansible_collections/arista/avd/plugins/module_utils/eos_designs_facts.py

```diff
@@ -320,16 +320,23 @@ def _switch_data(self):
         node_config = {}
         hostname = self.hostname
         node_type_config = get(self._hostvars, f"{self.node_type_key}", required=True)
-
-        if hostname in node_type_config.get('nodes', {}):
-            node_config = node_type_config['nodes'][hostname]
-        else:
-            for node_group in node_type_config.get('node_groups', {}):
-                if hostname in node_type_config['node_groups'][node_group].get('nodes', {}):
-                    node_config = node_type_config['node_groups'][node_group]['nodes'][hostname]
-                    switch_data['node_group'] = node_type_config['node_groups'][node_group]
-                    switch_data['group'] = node_group
-                    break
+        nodes = self._convert_dicts(node_type_config.get('nodes', []), 'name')
+
+        for node in nodes:
+            if hostname == node['name']:
+                node_config = node
```

Comment:
Note that technically here, you could end up having multiple `node[name]` being the same as `hostname` and selecting the last one. Below you are using `break` in the loop to stop on the first encounter in a node_group so slightly different behavior. Not sure what how we want to behave in this (misconfiguration) case but probably we should align.

Reply:
The schema will ensure that we have no duplicates. But I will add breaks here and under node_group level below for efficiency.

Reply:
done